### PR TITLE
use redux thunk and books api to fetch, delete and add books

### PR DESCRIPTION
**Pull request by sireclifford**
- use axios with createAsyncThunk
- Fetch book list from Bookstore API on Books page load.
- Make changes by using API.

Screenshot 2022-08-18 at 10 38 48 PM

Screenshot 2022-08-18 at 10 38 40 PM

### CODE REVIEW

File: src/components/addBook/AddBook.js [hunk 1]
```diff
@@ -1,46 +1,53 @@
...
       author: '',
     });
   };
 
+  useEffect(() => {
+    if (status === 'idle') {
+      dispatch(fetchBooks());
+    }
+  }, [status, dispatch]);
+
```
Comment (Reem-lab) [inline src/components/addBook/AddBook.js]: - [x] Great job on working on the functionality but I think there is a bug when adding a new book so after adding a book I can't see it till I refresh the page so each time I add a new book I should refresh the page so kindly fix it to add without refreshing ✨💯🚀🚀

**Review by Reem-lab — CHANGES_REQUESTED** [review]
# Hi @sireclifford, 👋🏻👋🏻

Good job so far! 🔥🔥🚀🚀
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights ✨✨
📌 No Linters error ✅
📌 Descriptive pull request✅
📌 Great working on the functionality ✅
📌 Good commit messages ✅
📌Professional README file. ✅
📌Very nice error message pop up ✅🔥💯



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/addBook/AddBook.js (hunk 1 above)
Comment (sireclifford) [reply src/components/addBook/AddBook.js:50]: @Reem-lab thank you for your review. As a matter of fact, I really don't know what to try again😩. I knew the problem but I had to submit it anyways. Would you happen to have an idea about how I can go about it?
The API doesn't return the rest of the books and so I don't know how to update the state.
Thank you🥺

Comment (sireclifford) [reply src/components/addBook/AddBook.js]: @Reem-lab thank you. Changes has been made

**Review by uwadonat — APPROVED** [review]
## Hi @sireclifford,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
### Highlights
- The project works as expected ✔️
- Good design ✔️
approval


# Status: Approved ✔️✔️✔️

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100%.
Please, remember to tag me @uwadonat in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**